sw/device/lib/crypto/impl/ecc/ecdsa_p256.h

```diff
@@ -59,6 +80,19 @@ status_t ecdsa_p256_keygen_finalize(p256_masked_scalar_t *private_key,
 status_t ecdsa_p256_sign_start(const uint32_t digest[kP256ScalarWords],
                                const p256_masked_scalar_t *private_key);
 
+/**
+ * Start an async ECDSA/P-256 signature generation operation on OTBN.
+ *
+ * Expects a sideloaded key from keymgr to be already loaded on OTBN. Returns
+ * an `OTCRYPTO_ASYNC_INCOMPLETE` error if OTBN is busy.
+ *
+ * @param digest Digest of the message to sign.
+ * @param private_key Secret key to sign the message with.
+ * @return Result of the operation (OK or error).
+ */
+status_t ecdsa_p256_sideload_sign_start(
+    const uint32_t digest[kP256ScalarWords]);
```

Comment:
is there a reason you pass some arrays as sized arrays and others as pointers?

Reply:
The digest is always a sized array, while elliptic-curve points and masked scalars are pointers. I could make a struct for the digest too, but that subjectively seemed superfluous to me because it would have only one field; the other structs all have at least two fields. But I don't object to changing it if it violates some kind of style best practice.

Reply:
No need for a struct, I was just curious why you didn't declare the function as `status_t ecdsa_p256_sideload_sign_start(const uint32_t *digest);` since even the sized array decays to a pointer in the function arg no? But if its for style reasons, best to keep things consistent as you have been :)